src/services/WalletService.ts

```diff
@@ -200,11 +183,60 @@ export default abstract class WalletService {
     }
   }
 
-  private static getUnusedValue(addressList: WalletAddress[]): Promise<WalletAddress[]> {
-    const addressListResponse = [...addressList];
+  private async askForBalance(sessionId: string, startFrom: number): Promise<void> {
+    const maxAddressPerCall: number = EnvironmentAccessorService
+      .getEnvironmentVariables().vueAppWalletAddressPerCall;
+    let addresses = await this.getAccountAddresses(maxAddressPerCall, startFrom);
+    if (addresses.length === 0) {
+      throw new Error('Error getting list of addresses - List of addresses is empty');
+    }
+    addresses = await this.getUnusedValue(addresses);
+    const balancesFound = await ApiService.getBalances(sessionId, addresses);
+    const balances = {
+      legacy: new SatoshiBig(balancesFound.legacy || 0, 'satoshi'),
+      segwit: new SatoshiBig(balancesFound.segwit || 0, 'satoshi'),
+      nativeSegwit: new SatoshiBig(balancesFound.nativeSegwit || 0, 'satoshi'),
+    };
+    if (balances) {
+      this.balanceAccumulated = {
+        legacy: new SatoshiBig(this.balanceAccumulated.legacy.plus(balances.legacy), 'satoshi'),
+        segwit: new SatoshiBig(this.balanceAccumulated.segwit.plus(balances.segwit), 'satoshi'),
+        nativeSegwit: new SatoshiBig(this.balanceAccumulated.nativeSegwit.plus(balances.nativeSegwit), 'satoshi'),
+      };
+      this.informSubscribers(this.balanceAccumulated, addresses);
+      return;
+    }
+    throw new Error('Error getting balances');
```

Comment:
Just a suggestion... invert the conditions to avoid having a throw as the linear flow :)
```suggestion
    if (!balances) {  
      throw new Error('Error getting balances');
   }
    this.balanceAccumulated = {
      legacy: new SatoshiBig(this.balanceAccumulated.legacy.plus(balances.legacy), 'satoshi'),
      segwit: new SatoshiBig(this.balanceAccumulated.segwit.plus(balances.segwit), 'satoshi'),
      nativeSegwit: new SatoshiBig(this.balanceAccumulated.nativeSegwit.plus(balances.nativeSegwit), 'satoshi'),
    };
    this.informSubscribers(this.balanceAccumulated, addresses);
```